exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -122,11 +122,25 @@ func timeSeriesSignature(metric *otlp.Metric, labels *[]prompb.Label) string {
 // createLabelSet creates a slice of Cortex Label with OTLP labels and paris of string values.
 // Unpaired string value is ignored. String pairs overwrites OTLP labels if collision happens, and the overwrite is
 // logged. Resultant label names are sanitized.
-func createLabelSet(labels []common.StringKeyValue, extras ...string) []prompb.Label {
+func createLabelSet(labels []common.StringKeyValue, externalLabels []ExternalLabel, extras ...string) []prompb.Label {
 
 	// map ensures no duplicate label name
 	l := map[string]prompb.Label{}
 
+	for _, el := range externalLabels {
+		name := el.Key
+		// The reserved prefix "__" is allowed in an external label
+		if name[:2] == "__" {
+			name = "__" + sanitize(name[2:])
```

Comment:
What's being sanitized here? I'm not sure if we should be altering the external labels as they act as a global override and just specify what's an allowed external label.

Reply:
This is how external labels are handled in prometheus. Some sanitization still must be done, for example if the external label entered by the user is "\_\_host.name\_\_", "." is not a valid character in a prometheus label

Reply:
That being said, external labels are not quite an override for all restrictions on a label, just for the "\_\_" restriction